src/main/java/ecs/PolygonCollider.java

```diff
@@ -0,0 +1,151 @@
+package ecs;
+
+import physics.collision.Collider;
+import physics.collision.CollisionInformation;
+import physics.collision.shape.PrimitiveShape;
+import util.MathUtils;
+
+/**
+ * A polygon collider for a {@link GameObject}.
+ * Collision detection uses GJK in {@link MathUtils#gjksmCollision(PrimitiveShape, PrimitiveShape)}.
+ * Must not be mixed with other colliders.
+ *
+ * @author Juyas
+ * @version 06.12.2021
+ * @since 06.12.2021
+ */
+public class PolygonCollider extends Component implements Collider {
+
+    /**
+     * The collision shape of the collider.
+     *
+     * @see PrimitiveShape
+     */
+    private PrimitiveShape shape;
+
+    /**
+     * A short representing a binary series of 0s and 1s.
+     * 0 means not present, 1 means present.
+     * Beeing present means, beeing present on the collision layer
+     * and allow collision with all objects having a present mask on this layer.
+     */
+    private short collisionLayer;
+
+    /**
+     * A short representing a binary series of 0s and 1s.
+     * 0 means not present, 1 means present.
+     * Beeing present means, beeing able to collide with all objects present on that layer,
+     * while its not required to be present on the layer.
+     */
+    private short collisionMask;
```

Comment:
Just something to think about, should the collision layer and mask be defined in Collider?

Reply:
yes